src/Cedar/CedarApplication.cs

```diff
@@ -1,40 +1,59 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Reflection;
-using System.Text;
-using System.Threading.Tasks;
-using System.Windows.Media.Imaging;
+﻿using System.Reflection;
+using Autodesk.Revit.ApplicationServices;
+using Autodesk.Revit.DB.Events;
 using Autodesk.Revit.UI;
+using Jpp.Cedar.Core;
+using Jpp.Cedar.Piling;
 
 namespace Jpp.Cedar
 {
     public class CedarApplication : IExternalApplication
     {
+        /// <summary>
+        /// Entry point for application
+        /// </summary>
+        /// <param name="application">Application being launched</param>
+        /// <returns>Result of add-in load</returns>
         public Result OnStartup(UIControlledApplication application)
         {
-            // Add a new ribbon panel
-            RibbonPanel ribbonPanel = application.CreateRibbonPanel("NewRibbonPanel");
+            if(application == null)
+                throw new System.ArgumentNullException(nameof(application));
```

Comment:
What's the impact on revit when throwing an exception here? Could we not just return Result.Failed?

Reply:
Not sure. Worth investigating, but seem to implcitly fail